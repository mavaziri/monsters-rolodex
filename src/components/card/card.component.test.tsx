// Card.component.test.tsx
import { render, screen } from '@testing-library/react';

import Card from './card.component';

const mockMonster = {
  id: '1',
  name: 'Monster One',
  email: 'monster1@example.com',
};

describe('Card Component', () => {
  test('renders the monster name', () => {
    render(<Card monster={mockMonster} />);

    expect(screen.getByText('Monster One')).toBeInTheDocument();
  });

  test('renders the monster email', () => {
    render(<Card monster={mockMonster} />);

    expect(screen.getByText('monster1@example.com')).toBeInTheDocument();
  });

  test('renders the monster image', () => {
    render(<Card monster={mockMonster} />);

    const imgElement = screen.getByRole('img', { name: /monster/i });
    expect(imgElement).toHaveAttribute(
      'src',
      'https://robohash.org/1?set=set2&size=180x180'
    );
  });
});
