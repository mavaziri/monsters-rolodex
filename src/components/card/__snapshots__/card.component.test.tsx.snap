// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Card component with monster details 1`] = `
<div>
  <div
    class="card-container"
  >
    <img
      alt="monster"
      src="https://robohash.org/1?set=set2&size=180x180"
    />
    <h2>
       
      Monster One
       
    </h2>
    <p>
      monster1@example.com
    </p>
  </div>
</div>
`;
